client/client.go

```diff
@@ -63,15 +63,21 @@ type client struct {
 }
 
 func New(cfg Config) (Client, error) {
```

Comment:
it seems best for the client library to receive the logging object. do you agree?

Reply:
I am not sure I agree. Everything is passed to the `Client` via the `config` object, why should the logger be an exception?

Reply:
I mean, the client is receiving a logLevel, not the logger

Reply:
Oh re-reading this I understand this better now, and yes, I agree.